vespalib/src/vespa/vespalib/btree/btreeiterator.h

```diff
@@ -451,6 +483,54 @@ class BTreeIteratorBase
             _leafRoot->foreach_key(func);
         }
     }
+
+    template <typename FunctionType>
+    void
+    foreach_key_range(const BTreeIteratorBase &end_itr, FunctionType func) const
+    {
+        if (!valid()) {
+            return;
+        }
+        if (!end_itr.valid()) {
+            foreach_key_range_start(_pathSize, func);
+            return;
+        }
+        assert(_pathSize == end_itr._pathSize);
+        assert(_allocator == end_itr._allocator);
+        uint32_t level = _pathSize;
+        if (level > 0u) {
+            uint32_t idx;
+            uint32_t eidx;
+            do {
```

Comment:
Please add a comment on what this the result of this while loop.

Reply:
Done